x-pack/platform/plugins/shared/streams/server/lib/streams/helpers/migrate_to_streamlang_on_read.ts

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  ALWAYS_CONDITION,
+  Condition,
+  ManualIngestPipelineProcessor,
+  StreamlangDSL,
+  conditionToPainless,
+} from '@kbn/streamlang';
+
+export const migrateRoutingIfConditionToStreamlang = (definition: Record<string, unknown>) => {
```

Comment:
can do in a follow-up, but should we have some tests for this?

Reply:
Yep, there's already some for the processing array via the migrate on read tests, I'll extend to cover this.